Drift/Private/DriftBase.cpp

```diff
@@ -4639,6 +4654,62 @@ void FDriftBase::GetUserIdentitiesByName(const FString& name, const FDriftGetUse
     GetUserIdentitiesByNames(namesArray, delegate);
 }
 
+void FDriftBase::CacheFriendRichPresence(int32 FriendId, const FDriftGetFriendRichPresenceDelegate& Delegate)
+{
+    if (state_ != DriftSessionState::Connected)
+    {
+        DRIFT_LOG(Base, Warning, TEXT("Attempting to get rich presence without being connected"));
+        (void) Delegate.ExecuteIfBound(false, {});
+        return;
+    }
+
+    FString Url = driftEndpoints.template_richpresence.Replace(TEXT("{player_id}"), *FString::FromInt(FriendId));
+
+    const auto Request = GetGameRequestManager()->Get(Url);
+    Request->OnResponse.BindLambda([this, Delegate, FriendId](ResponseContext& Context, JsonDocument& Doc)
+    {
+        FRichPresenceResult Result;
+        if (!JsonArchive::LoadObject(Doc, Result))
+        {
+            Context.error = TEXT("Failed to parse user identities response");
+            return;
+        }
+
+        RichPresenceCache.Add(FriendId, Result);
+        (void) Delegate.ExecuteIfBound(true, Result);
+    });
+
+    Request->OnError.BindLambda([this, Delegate, Url](ResponseContext& Context)
+    {
+        FString Error;
+        Context.errorHandled = GetResponseError(Context, Error);
+        const FString ErrorMessage = FString::Printf(TEXT("Failed to get richpresence from url: %s. Error: '%s'"), *Url, *Error);
+        DRIFT_LOG(Base, Error, TEXT("%s"), *ErrorMessage);
+        IErrorReporter::Get()->AddError(TEXT("FDriftBase"), *ErrorMessage);
+    });
+}
+
+void FDriftBase::CacheFriendsRichPresence(const FDriftGetFriendsRichPresenceDelegate& Delegate)
+{
+    TArray<FDriftFriend> DriftFriends;
+    GetFriendsList(DriftFriends);
+    const int32 NumFriends = DriftFriends.Num();
+    int32 Results = 0;
+
+    for (auto& DriftFriend : DriftFriends)
+    {
+        CacheFriendRichPresence(DriftFriend.playerID, FDriftGetFriendRichPresenceDelegate::CreateLambda([&Results, NumFriends, Delegate](bool Success, const FRichPresenceResult& Result)
```

Comment:
There's an issue here, the lambda is capturing the local variable `Results` by reference but CacheFriendRichPresence is async